x-pack/plugins/security_solution/server/assistant/tools/open_and_acknowledged_alerts/open_and_acknowledged_alerts_tool.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import { getAnonymizedValue, transformRawData } from '@kbn/elastic-assistant-common';
+import { DynamicTool } from 'langchain/tools';
+import { requestHasRequiredAnonymizationParams } from '@kbn/elastic-assistant-plugin/server/lib/langchain/helpers';
+
+import type { AssistantTool, AssistantToolParams } from '@kbn/elastic-assistant-plugin/server';
+import { getOpenAndAcknowledgedAlertsQuery } from './get_open_and_acknowledged_alerts_query';
+import { getRawDataOrDefault, sizeIsOutOfRange } from './helpers';
+import { APP_ID } from '../../../../common';
+
+export interface OpenAndAcknowledgedAlertsToolParams extends AssistantToolParams {
+  alertsIndexPattern: string;
+  size: number;
+}
+
+export const OPEN_AND_ACKNOWLEDGED_ALERTS_TOOL_DESCRIPTION =
+  'Call this for knowledge about the latest n open and acknowledged alerts (sorted by `kibana.alert.risk_score`) in the environment, or when answering questions about open alerts';
```

Comment:
thanks for adding `acknowledged` to the tool description [per this comment in #173121](https://github.com/elastic/kibana/pull/173121#discussion_r1425748603) 🙏